src/document/mod.rs

```diff
@@ -84,113 +116,225 @@ impl DocumentEncryptOpts {
     ///
-    /// # Returns
-    /// `Result<DocumentId>` Fails if provided encrypted document has no header, otherwise returns extracted ID.
+    /// # Examples
+    /// ```
+    /// # fn run() -> Result<(), ironoxide::IronOxideErr> {
+    /// # use ironoxide::prelude::*;
+    /// # let sdk: IronOxide = unimplemented!();
+    /// # let bytes: Vec<u8> = vec![];
+    /// let document_id = sdk.document_get_id_from_bytes(&bytes)?;
+    /// # Ok(())
+    /// # }
+    /// ```
     fn document_get_id_from_bytes(&self, encrypted_document: &[u8]) -> Result<DocumentId>;
 
-    /// Encrypt the provided document bytes.
+    /// Updates the contents of an existing IronCore encrypted document.
     ///
-    /// # Arguments
-    /// - `document_data` - Bytes of the document to encrypt
-    /// - `encrypt_opts` - Optional document encrypt parameters. Includes
-    ///       `id` - Unique ID to use for the document. Document ID will be stored unencrypted and must be unique per segment.
-    ///       `name` - Non-unique name to use in the document. Document name will **not** be encrypted.
-    ///       `grant_to_author` - Flag determining whether to encrypt to the calling user or not. If set to false at least one value must be present in the `grant` list.
-    ///       `grants` - List of users/groups to grant access to this document once encrypted
-    async fn document_encrypt(
-        &self,
-        document_data: &[u8],
-        encrypt_opts: &DocumentEncryptOpts,
-    ) -> Result<DocumentEncryptResult>;
-
-    /// Update the encrypted content of an existing document. Persists any existing access to other users and groups.
+    /// The new contents will be encrypted, and which users and groups are granted access
+    /// will remain unchanged.
     ///
     /// # Arguments
-    /// - `id` - ID of document to update.
-    /// - `new_document_data` - Updated document content to encrypt.
+    /// - `id` - Unique ID of the document to update
+    /// - `new_document_data` - Updated bytes to encrypt
+    ///
+    /// # Examples
+    /// ```
+    /// # async fn run() -> Result<(), ironoxide::IronOxideErr> {
+    /// # use ironoxide::prelude::*;
+    /// # let sdk: IronOxide = unimplemented!();
+    /// # let document_id: DocumentId = unimplemented!();
+    /// let new_data = "more secret data".as_bytes();
+    /// let encrypted = sdk.document_update_bytes(&document_id, new_data).await?;
+    /// # Ok(())
+    /// # }
+    /// ```
     async fn document_update_bytes(
         &self,
         id: &DocumentId,
         new_document_data: &[u8],
     ) -> Result<DocumentEncryptResult>;
 
-    /// Decrypts the provided encrypted document and returns details about the document as well as its decrypted bytes.
-    ///
-    /// # Arguments
-    /// - `encrypted_document` - Bytes of encrypted document. Should be the same bytes returned from `document_encrypt`.
+    /// Updates a document's name to a new value.
```

Comment:
```suggestion
    /// Modify or remove a document's name.
```

Reply:
I like it, though I'll change the verb tenses to match the standard/consistency.